prrte build fails when lsf component is enabled

**Version**: prrte master

Here is the failure log:

```
ras_lsf_module.c:124:43: error: ‘prte_hwloc_binding_policy’ undeclared (first use in this function)
                PRTE_BINDING_POLICY_IS_SET(prte_hwloc_binding_policy)) {
                                           ^
/workspace/x86_64/ompi-73/build/ompi/prrte/src/hwloc/hwloc-internal.h:167:7: note: in definition of macro ‘PRTE_BINDING_POLICY_IS_SET’
     ((pol) & 0x4000)
       ^
ras_lsf_module.c:124:43: note: each undeclared identifier is reported only once for each function it appears in
                PRTE_BINDING_POLICY_IS_SET(prte_hwloc_binding_policy)) {
                                           ^
/workspace/x86_64/ompi-73/build/ompi/prrte/src/hwloc/hwloc-internal.h:167:7: note: in definition of macro ‘PRTE_BINDING_POLICY_IS_SET’
     ((pol) & 0x4000)
       ^
ras_lsf_module.c:149:48: error: ‘PRTE_JOB_PHYSICAL_CPUIDS’ undeclared (first use in this function)
         prte_set_attribute(&jdata->attributes, PRTE_JOB_PHYSICAL_CPUIDS, true, NULL, PRTE_BOOL);
                                                ^
ras_lsf_module.c:151:9: error: ‘prte_hwloc_use_hwthreads_as_cpus’ undeclared (first use in this function)
         prte_hwloc_use_hwthreads_as_cpus = true;
         ^
```

Looks like the changes in this PR is not ported to ras/lsf compoent.
https://github.com/openpmix/prrte/pull/555/files

--- a/src/mca/plm/lsf/plm_lsf_module.c
+++ b/src/mca/plm/lsf/plm_lsf_module.c
@@ -388,7 +388,7 @@ static void launch_daemons(int fd, short args, void *cbdata)
 
     /* check for failed launch - if so, force terminate */
     if (failed_launch) {
-        PRTE_ACTIVATE_JOB_STATE(state->jata, PRTE_JOB_STATE_FAILED_TO_START);
+        PRTE_ACTIVATE_JOB_STATE(state->jdata, PRTE_JOB_STATE_FAILED_TO_START);
     }
 
     /* cleanup the caddy */

--- a/src/mca/ras/lsf/help-ras-lsf.txt
+++ b/src/mca/ras/lsf/help-ras-lsf.txt
@@ -35,4 +35,13 @@ The affinity file provided in LSB_AFFINITY_HOSTFILE could not be found:
   File:  %s
 
 We cannot continue.
+[affinity-file-found-not-used]
+The affinity file provided in LSB_AFFINITY_HOSTFILE could not be used:
 
+  File:   %s
+  Reason: %s
+
+Use the MCA parameter below to skip processing the LSB_AFFINITY_HOSTFILE
+and ignore any binding provided by LSF for this job. Doing so may
+require you to use the affinity options provided by PRTE.
+  --mca ras_lsf_skip_affinity_file true

--- a/src/mca/ras/lsf/ras_lsf.h
+++ b/src/mca/ras/lsf/ras_lsf.h
@@ -38,6 +38,8 @@ BEGIN_C_DECLS
 PRTE_EXPORT extern prte_ras_base_component_t prte_ras_lsf_component;
 PRTE_EXPORT extern prte_ras_base_module_t prte_ras_lsf_module;
 
+PRTE_EXPORT extern bool prte_ras_lsf_skip_affinity_file;
+
 END_C_DECLS
 
 #endif

--- a/src/mca/ras/lsf/ras_lsf_component.c
+++ b/src/mca/ras/lsf/ras_lsf_component.c
@@ -39,7 +39,9 @@
 static int prte_ras_lsf_open(void);
 static int prte_ras_lsf_close(void);
 static int prte_ras_lsf_component_query(prte_mca_base_module_t **module, int *priority);
+static int prte_ras_lsf_register(void);
 
+bool prte_ras_lsf_skip_affinity_file = false;
 
 prte_ras_base_component_t prte_ras_lsf_component = {
     .base_version = {
@@ -54,6 +56,7 @@ prte_ras_base_component_t prte_ras_lsf_component = {
         .mca_open_component = prte_ras_lsf_open,
         .mca_close_component = prte_ras_lsf_close,
         .mca_query_component = prte_ras_lsf_component_query,
+        .mca_register_component_params = prte_ras_lsf_register,
     },
     .base_data = {
         /* The component is checkpoint ready */
@@ -94,4 +97,17 @@ static int prte_ras_lsf_close(void)
     return PRTE_SUCCESS;
 }
 
+static int prte_ras_lsf_register(void)
+{
+    prte_ras_lsf_skip_affinity_file = false;
+    (void) prte_mca_base_component_var_register(&prte_ras_lsf_component.base_version,
+                                                "skip_affinity_file",
+                                                "Skip processing the LSB_AFFINITY_HOSTFILE.",
+                                                PRTE_MCA_BASE_VAR_TYPE_BOOL, NULL,
+                                                0, PRTE_MCA_BASE_VAR_FLAG_NONE,
+                                                PRTE_INFO_LVL_3,
+                                                PRTE_MCA_BASE_VAR_SCOPE_READONLY,
+                                                &prte_ras_lsf_skip_affinity_file);
 
+    return PRTE_SUCCESS;
+}

--- a/src/mca/ras/lsf/ras_lsf_module.c
+++ b/src/mca/ras/lsf/ras_lsf_module.c
@@ -122,12 +122,14 @@ static int allocate(prte_job_t *jdata, prte_list_t *nodes)
             directives_given = true;
         }
     } else if ((PRTE_MAPPING_GIVEN & PRTE_GET_MAPPING_DIRECTIVE(prte_rmaps_base.mapping)) ||
-               PRTE_BINDING_POLICY_IS_SET(prte_hwloc_binding_policy)) {
+               PRTE_BINDING_POLICY_IS_SET(prte_hwloc_default_binding_policy)) {
             directives_given = true;
     }
 
     /* check for an affinity file */
-    if (!directives_given && NULL != (affinity_file = getenv("LSB_AFFINITY_HOSTFILE"))) {
+    if (!prte_ras_lsf_skip_affinity_file &&
+        !directives_given &&
+        NULL != (affinity_file = getenv("LSB_AFFINITY_HOSTFILE"))) {
         /* check to see if the file is empty - if it is,
          * then affinity wasn't actually set for this job */
         if (0 != stat(affinity_file, &buf)) {
@@ -138,6 +140,17 @@ static int allocate(prte_job_t *jdata, prte_list_t *nodes)
             /* no affinity, so just return */
             return PRTE_SUCCESS;
         }
+#if 1
+        // Phsical CPU IDs are no longer supported. See the Issue below:
+        //   https://github.com/openpmix/prrte/issues/791
+        // Until that is resolved throw an error if we detect that the user is
+        // trying to use LSF level affinity options.
+        if( NULL != affinity_file ) { // Always true
+            prte_show_help("help-ras-lsf.txt", "affinity-file-found-not-used", true,
+                           affinity_file, "Physical CPU ID mapping is not supported");
+            return PRTE_ERR_SILENT;
+        }
+#else
         /* the affinity file sequentially lists rank locations, with
          * cpusets given as physical cpu-ids. Setup the job object
          * so it knows to process this accordingly */
@@ -147,12 +160,14 @@ static int allocate(prte_job_t *jdata, prte_list_t *nodes)
         PRTE_SET_MAPPING_POLICY(jdata->map->mapping, PRTE_MAPPING_SEQ);
         jdata->map->req_mapper = strdup("seq"); // need sequential mapper
         /* tell the sequential mapper that all cpusets are to be treated as "physical" */
-        prte_set_attribute(&jdata->attributes, PRTE_JOB_PHYSICAL_CPUIDS, true, NULL, PMIX_BOOL);
+        // TODO - Physical CPUs are no longer supported by PRRTE. Need a fix the following
+        //        attribute is no longer valid.
+        //prte_set_attribute(&jdata->attributes, PRTE_JOB_PHYSICAL_CPUIDS, true, NULL, PMIX_BOOL);
         /* LSF provides its info as hwthreads, so set the hwthread-as-cpus flag */
-        prte_hwloc_use_hwthreads_as_cpus = true;
+        prte_set_attribute(&jdata->attributes, PRTE_JOB_HWT_CPUS, true, NULL, PMIX_BOOL);
         /* don't override something provided by the user, but default to bind-to hwthread */
-        if (!PRTE_BINDING_POLICY_IS_SET(prte_hwloc_binding_policy)) {
-            PRTE_SET_BINDING_POLICY(prte_hwloc_binding_policy, PRTE_BIND_TO_HWTHREAD);
+        if (!PRTE_BINDING_POLICY_IS_SET(prte_hwloc_default_binding_policy)) {
+            PRTE_SET_BINDING_POLICY(prte_hwloc_default_binding_policy, PRTE_BIND_TO_HWTHREAD);
         }
         /*
          * Do not set the hostfile attribute on each app_context since that
@@ -168,7 +183,7 @@ static int allocate(prte_job_t *jdata, prte_list_t *nodes)
         prte_default_hostfile = strdup(affinity_file);
         prte_output_verbose(10, prte_ras_base_framework.framework_output,
                             "ras/lsf: Set default_hostfile to %s",prte_default_hostfile);
-
+#endif
         return PRTE_SUCCESS;
     }
 

